osu.Game/Beatmaps/Drawables/Cards/Statistics/BeatmapCardDateStatistic.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System;
+using osu.Framework.Extensions.LocalisationExtensions;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+
+namespace osu.Game.Beatmaps.Drawables.Cards.Statistics
+{
+    public class BeatmapCardDateStatistic : BeatmapCardStatistic
+    {
+        private readonly DateTimeOffset dateTime;
+
+        private BeatmapCardDateStatistic(DateTimeOffset dateTime)
+        {
+            this.dateTime = dateTime;
+
+            Icon = FontAwesome.Regular.CheckCircle;
+            Text = dateTime.ToLocalisableString(@"d MMM yyyy");
+        }
+
+        public override object TooltipContent => dateTime;
+        public override ITooltip GetCustomTooltip() => new DateTooltip();
+
+        public static BeatmapCardDateStatistic? CreateFor(IBeatmapSetOnlineInfo beatmapSetInfo)
```

Comment:
Hmm, I can see why this was required, but not sure about the structure. I guess the reasoning for putting it here is that this component is going to be reused in other places and placing the `displayDateFor` code in `BeatmapCard` would disallow easy reuse?

Reply:
Right. If the appropriate date is null in the API response I somehow need to not show anything at all. There are many ways to do this, most of which are shaped something like this, though.

I could replace `null` with `Empty()`, or I could try setting alpha to 0 in such cases - the second option may be a bit WTF if someone happens to need to debug it, though.

Reply:
I think it's alright, just a different pattern to what we've done until now. We'll see if it sticks.